torchrl/collectors/collectors.py

```diff
@@ -81,6 +83,15 @@ def recursive_map_to_cpu(dictionary: OrderedDict) -> OrderedDict:
     )
 
 
+def _module_is_tensordict_compatible(module: nn.Module):
+    sig = inspect.signature(module.forward)
```

Comment:
I guess I was more thinking of this

a function that returns True in these cases
1. 
```
class MyModule(nn.Module):
    in_keys = ["stuff"]
    out_keys = ["other_stuff"]
    def forward(self, tensordict):
        pass
```
2. TensorDictModule

--- 
returns False in those cases
1. 
```
class MyModule(nn.Module):
    in_keys = ["stuff"]
    out_keys = ["other_stuff"]
    def forward(self, tensordict, tensor):  # tensor is unexpected here
        pass
```

```
class MyModule(nn.Module):
    def forward(self, tensor):
        pass
```

--- 
We could also be super restrictive
Return True iif one of:
- TensorDictmodule
- nn.Module + in_keys + out_keys + one input only (assumed to be of type tensordict)

Return False iif
- _no_ in_keys / out_keys attribute and whatever input

if there are in_keys and/or out_keys but there are multiple inputs or similar, it's likely that the use is trying to do smth that will have an undetermined behaviour.

Other cases: throw an error

(in_keys / out_keys is an indirect indicator of "smth that works with TensorDict" which is always worse than a direct indicator but I guess that's the best we can do right now)